reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -214,6 +219,9 @@ def iterative_refinement(self, count_norm_const=1):
                     )
                 )
 
+                log_post_1 += em_loss_1
```

Comment:
you don't add it between iterations - just log it fresh each round. each time it's an evaluation of the loss, and doesn't accumulate between iterations.